spec/models/post_spec.rb

```diff
@@ -0,0 +1,43 @@
+require 'rails_helper'
+
+RSpec.describe Post, type: :model do
+  let(:user) { User.create(name: 'John', posts_counter: 0) }
+
+  it 'is valid with a title, comments_counter, and likes_counter' do
+    post = user.posts.new(title: 'Example Post', comments_counter: 0, likes_counter: 0)
+    expect(post).to be_valid
+  end
+
+  it 'is invalid without a title' do
+    post = user.posts.new(comments_counter: 0, likes_counter: 0)
+    expect(post).to_not be_valid
+  end
+
+  it 'is invalid with a title exceeding 250 characters' do
+    long_title = 'a' * 251
+    post = user.posts.new(title: long_title, comments_counter: 0, likes_counter: 0)
+    expect(post).to_not be_valid
+  end
+
+  it 'updates the posts_counter for the author' do
+    post = user.posts.create(title: 'Example Post', comments_counter: 0, likes_counter: 0)
+    expect { post.update_post_counter }.to change { user.reload.posts_counter }.by(1)
+  end
+  # method tests
+  it 'should test update_user_counter' do
+    user = User.new(name: 'najib', posts_counter: 0)
+    post = Post.create(title: 'Example Post', author: user)
+    expect(user.posts_counter).to eq(0)
+    post.update_post_counter
+    expect(user.posts_counter).to eq(1)
+  end
+
+  it 'should test recent comments  method' do
+    user = User.new(name: 'najib', posts_counter: 0)
+    post = Post.create(title: 'Example Post', author: user)
+    comments = []
+    recent_comments = post.recent_comments
+    expect(recent_comments).to eq(comments.reverse)
+    expect(recent_comments.count).to eq(0)
+  end
```

Comment:
- [ ] Just as the previous reviewer mentioned, Kindly test your `recent _comment` method to ensure that the method works as expected but this test doesn't quite do that. Kindly refer to the [test example](https://github.com/najibullahjafari/ruby_social_media/pull/3/files#r1349505163) the previous reviewer provided you with 👍 
- [ ] Your test needs to check that the `recent_comments` method actually returns 5 recent comments 👍 
- [ ] - Your method
![image](https://github.com/najibullahjafari/ruby_social_media/assets/102699958/9a0338b3-ccce-4410-9cb1-1a3c937675ba)